File: app/src/main/java/com/jeronimo/swingradio/utils/FragmentsController.java
package com.jeronimo.swingradio.utils;

import android.app.Fragment;
import android.app.FragmentManager;
import android.app.FragmentTransaction;

/**
 * Created by gleb on 23.12.15.
 */
public class FragmentsController {

    public static final String MAIN_STACK = "main";
    private static  FragmentsController instance;
    private FragmentManager mFragmentManager;
    private int mContainerID;


    public static FragmentsController getInstance()
    {
        if(instance == null)
        {
            instance = new FragmentsController();
        }
        return  instance;
    }

    public void init(FragmentManager fragmentManager)
    {
        mFragmentManager = fragmentManager;
    }

    public void setContainer(int id)
    {
        mContainerID = id;
    }

    public void addFragmentToContainer(Fragment fragment, int container)
    {
        FragmentTransaction fragmentTransaction = mFragmentManager.beginTransaction();
        fragmentTransaction.add(container, fragment, getTagForFragment(fragment));
        fragmentTransaction.addToBackStack(MAIN_STACK);

        fragmentTransaction.commit();

    }

    public void backInStack()
    {
        mFragmentManager.popBackStack(MAIN_STACK, 0);
    }

    public static String getTagForFragment(Fragment fragment)
    {
        return fragment.getClass().getName();
    }


}

File: app/src/main/java/com/jeronimo/swingradio/utils/FragmentsController.kt
package com.jeronimo.swingradio.utils

import android.app.Fragment
import android.app.FragmentManager

/**
 * Created by gleb on 23.12.15.
 */
class FragmentsController {
    private var mFragmentManager: FragmentManager? = null
    private var mContainerID: Int = 0
    private object Holder { val INSTANCE = FragmentsController() }

    fun init(fragmentManager: FragmentManager) {
        mFragmentManager = fragmentManager
    }

    fun setContainer(id: Int) {
        mContainerID = id
    }

    fun addFragmentToContainer(fragment: Fragment, container: Int) {
        val fragmentTransaction = mFragmentManager!!.beginTransaction()
        fragmentTransaction.add(container, fragment, getTagForFragment(fragment))
        fragmentTransaction.addToBackStack(MAIN_STACK)

        fragmentTransaction.commit()

    }

    fun backInStack() {
        mFragmentManager!!.popBackStack(MAIN_STACK, 0)
    }

    companion object {

        val MAIN_STACK = "main"
        val instance : FragmentsController by lazy {Holder.INSTANCE}

        fun getTagForFragment(fragment: Fragment): String {
            return fragment.javaClass.name
        }
    }


}
